test/utils/shallowEquality.spec.js

```diff
@@ -0,0 +1,71 @@
+import expect from 'expect';
+import shallowEqualScalar from '../../src/utils/shallowEqualScalar';
+import shallowEqual from '../../src/utils/shallowEqual';
+
+describe('Utils', () => {
+  // More info: https://github.com/gaearon/redux/pull/75#issuecomment-111635748
+  describe('shallowEqualScalar', () => {
+    it('should compare the given objects shallowly', () => {
+
+      expect(shallowEqualScalar(null, {})).toBe(false);
+      expect(shallowEqualScalar({}, null)).toBe(false);
+      const m = {one: 1};
+
+      expect(shallowEqualScalar([], [])).toBe(true);
+      expect(shallowEqualScalar([], [1, 2])).toBe(false);
+      expect(shallowEqualScalar([3, 4], [3, 4])).toBe(true);
+      expect(shallowEqualScalar([3, 4], [3, 5])).toBe(false);
+      expect(shallowEqualScalar([[]], [[]])).toBe(false);
+      expect(shallowEqualScalar([m], [m])).toBe(false);
+
```

Comment:
There's a lot of tests for the arrays, but I'd expect to see more tests with objects.
For some inspiration: https://github.com/facebook/react/blob/master/src/shared/utils/__tests__/shallowEqual-test.js